cmake/modules/FindICU.cmake

```diff
@@ -2,28 +2,57 @@
 
 include(FindPackageHandleStandardArgs)
 
-find_package(PkgConfig)
-
 set(ICU_REQUIRED)
-foreach(MODULE ${ICU_FIND_COMPONENTS})
-  string(TOUPPER "${MODULE}" MODULE)
-  string(TOLOWER "${MODULE}" module)
-  list(APPEND ICU_REQUIRED
-    ICU_${MODULE}_INCLUDE_DIR ICU_${MODULE}_LIBRARIES)
-
-  pkg_check_modules(PC_ICU_${MODULE} QUIET icu-${module})
-  if(${PC_ICU_${MODULE}_FOUND})
-    set(ICU_${MODULE}_DEFINITIONS ${PC_ICU_${MODULE}_CFLAGS_OTHER})
-
-    find_path(ICU_${MODULE}_INCLUDE_DIR unicode
-      HINTS ${PC_ICU_${MODULE}_INCLUDEDIR} ${PC_ICU_${MODULE}_INCLUDE_DIRS})
-    set(ICU_${MODULE}_INCLUDE_DIR ${ICU_${MODULE}_INCLUDE_DIR})
-
-    find_library(ICU_${MODULE}_LIBRARY NAMES icu${module}
-      HINTS ${PC_ICU_${MODULE}_LIBDIR} ${PC_ICU_${MODULE}_LIBRARY_DIRS})
-    set(ICU_${MODULE}_LIBRARIES ${ICU_${MODULE}_LIBRARY})
+
+if ("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
```

Comment:
I think you're still not approaching this correctly.

Instead of having the `if ("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")`, why not allow the ICU paths to be overridden on all platforms? Then the PkgConfig code path can run whenever an override value isn't specified on any platform, and raise errors if it fails.

It is worth noting that the PkgConfig functions in CMake return not-found if the user doesn't have pkg-config installed, so you don't need to avoid calling those functions on Windows (or any other platform).